src/core/public/application/types.ts

```diff
@@ -192,6 +198,19 @@ export type AppUnmount = () => void;
 /** @internal */
 export type AppMounter = (params: AppMountParameters) => Promise<AppUnmount>;
 
+/** @internal */
+export type LegacyAppMounter = (params: AppMountParameters) => void;
+
+/** @internal */
+export type Mounter<T = App | LegacyApp> = SelectivePartial<
```

Comment:
Wouldn't it be simpler if we split it into 2 different types?

Reply:
If we split this into different types, then it means another layer branching in order to render the application from the UI. Going this route will still be future-proof, while keeping the other logic simpler.

Reply:
If we split this into different types, then it means another layer branching in order to render the application from the UI. Going this route will still be future-proof, while keeping the other logic simpler.